### DISCUSSION TITLE
Creating a custom route based on createFileRoute?

### DISCUSSION THREAD

**Discussion by ThanMatt** [TanStack/router#2865] [Q&A] [answered]
Hello, I'm trying to create a custom route called `createProtectedRoute` that is based on `createFileRoute`. Here's my current implementation:
```typescript
export const createProtectedRoute = (
  path: keyof FileRoutesByPath,
  Component: RouteComponent,
  // :: additional args
) => {
  return createFileRoute(path)({
    component: Component,
    beforeLoad: ({ location }) => {
      const auth = useAuthStore.getState();
      if (!auth.accessToken) {
        throw redirect({
          to: "/login",
          search: {
            redirect: location.href,
          },
        });
      }
    },
    // :: Add more args here
  });
};

// :: Sample implementation
export const Route = createProtectedRoute("/", HomeComponent);
```
While this works, I'd like to make it more flexible by adding support for additional arguments like `loader.` How can I achieve this with typescript? Are there any alternative approaches you'd recommend?
Thank you!

**Comment by schiller-manuel** [comment, marked as answer]
this won't work well with automatic codesplitting unfortunately. we have this in our backlog but not planned yet.
